assets/fetchBook.js

```diff
@@ -0,0 +1,74 @@
+const url="https://www.googleapis.com/books/v1/volumes?q=";
+
+//immediately invoked function expression 
+(async()=>{
+    fetch(`${url}genre`)
+    .then(res=>res.json())
+    .then(data=>{
+        const correctData=transformData(data.items);
+        if(correctData){
+            books=correctData;            
+            localStorage.setItem('data',JSON.stringify(correctData));
+            updateBook(books);
+        }
+    })
+    .catch(err=>console.error(err));
+})();
+transformData=(apiData)=>{
+    console.log("Books from fetch",apiData);
+
+    return apiData.filter(
+        (data)=>
+        data.volumeInfo.title && 
+        data.volumeInfo.authors && 
+        data.volumeInfo.categories && 
+        data.volumeInfo.industryIdentifiers &&
+        data.volumeInfo.publishedDate
+        
+    ).map(
+        (data)=>({
+            title:data.volumeInfo.title,
+            author:data.volumeInfo.authors[0],
+            genre:data.volumeInfo.categories[0].toLowerCase(),
+            isbn:data.volumeInfo.industryIdentifiers[0].identifier,
+            date:data.volumeInfo.publishedDate
+        })
+    )
+}
+searchBook=async()=>{
+    const searchValue=document.getElementById('search').value.toLowerCase();
+    
+    //if search input is empty show all the books
+    if(searchValue===''){
+        updateBook(books);
+    }
+    else{
+        try {
+            const res=await fetch(`${url}title:${searchValue}`)
+            const data=await res.json();
+            if(res.ok){
+                //filter the date that includes particular title
+                const filterData=data.items.filter(d=>{
```

Comment:
here instead since you have all the data loaded you should filter that data rather than again making an API call